### ISSUE TITLE
Dashboard table sorting is case sensitive

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#11338]
**Describe the bug**
Dashboard table sorting uses case sensitive instead of case insensitive ordering, which makes all uppercase strings come before all lowercase.

**To Reproduce**
1. Create SQL question and save it
```
SELECT 1 AS "ID", 'abc' AS "Sort" UNION ALL
SELECT 2 AS "ID", 'ABC' AS "Sort" UNION ALL
SELECT 3 AS "ID", 'def' AS "Sort" UNION ALL
SELECT 4 AS "ID", 'DEF' AS "Sort" UNION ALL
SELECT 5 AS "ID", 'xyz' AS "Sort" UNION ALL
SELECT 6 AS "ID", 'XYZ' AS "Sort"
```
2. Add question to dashboard and Save.
3. Click "Sort" column (ascending)
4. Shows case sensitive sorting
```
ABC
DEF
XYZ
abc
def
xyz
```

**Expected behavior**
To use case insensitive sorting:
```
abc
ABC
def
DEF
xyz
XYZ
```

**Information about your Metabase Installation:**
Metabase 0.33.5.1 on Firefox 70

**Additional context**
https://discourse.metabase.com/t/need-case-insensitive-ordering-by-name-in-dashboard/8036

**Comment by emilythacher** [comment]
@flamber was this resolved? I am also observing case-sensitive sorting. I have a list of locations that all start with capital letters, including "M", and then two locations that start with lowercase "m". The two locations that start with "m" show up at the bottom of the list in: 

-  Table view of a question
- Bar chart view of a question
- Filter menu in question view
- Drop-down field filter on a dashboard

I have observed this in both Chrome and Firefox. We're running v0.35.2.

This is very inconvenient because the list is long and users sometimes have trouble finding the two "m" facilities since they don't show up with the other "M"s.

**Comment by flamber** [comment]
@emilythacher This was implemented since 0.34.0, but this was a frontend sorting issue.
It's not going to change the order of your data - that depends on your query and your database.
I would guess that your locations might contain spaces before the letters, which is what is causing the different order. You can try to download as CSV and check the location-column.
If that doesn't work, then please use the forum for troubleshooting: https://discourse.metabase.com/

**Comment by JamesPaden** [comment]
This problem is back. I was able to recreate it with the above steps.

**Comment by flamber** [comment]
And fixed again #20781

**Comment by JamesPaden** [comment]
Awesome, sorry i missed those!
